File: YinLu/HtmlUnit_qgTest/src/qgtest/Scene5.java
package qgtest;

import java.util.Set;

import org.testng.Assert;
import org.testng.annotations.Test;

import com.gargoylesoftware.htmlunit.util.Cookie;

import net.sf.json.JSONObject;

public class Scene5 {
//未登录，无收货地址
	@Test(priority=0)
	public void login() throws Exception {
		String url="http://study-perf.qa.netease.com/common/fgadmin/login";
		JSONObject login = new JSONObject();
		login.element("password", "netease123");
		login.element("phoneArea", "86");
		login.element("phoneNumber", "200000000055");
		String result=HtmlUnit.doPost(url, login);
		System.out.println("login_success"+result);
	}
	@Test(priority=1)
	public void addresslist() throws Exception {

		String result=HtmlUnit.doGet("http://study-perf.qa.netease.com/fgadmin/address/list");
		System.out.println(result);
		JSONObject json=JSONObject.fromObject(result);
		Assert.assertEquals(json.getString("message"), "success");
		
	}
	@Test(priority=2)
	public void addressnew() throws Exception {
		String url="http://study-perf.qa.netease.com/fgadmin/address/new";

		JSONObject address = new JSONObject();
		address.element("receiverName", "李四");
		address.element("cellPhone", "20000000004");
		address.element("addressDetail", "南京大学");
		address.element("province", "江苏省");
		address.element("city", "南京市");
		address.element("area", "鼓楼区");
		String result=HtmlUnit.doPost(url, address);	
		System.out.println(result);
	}
	@Test(priority=3)
	public void fee() throws Exception{
		String url="http://study-perf.qa.netease.com/common/getTransportFee?id=1&addressDetail=江苏省_南京市_鼓楼区";
		String result=HtmlUnit.doGet(url);
		System.out.println(result);
	}
	@Test(priority=4)
	public void submit() throws Exception {
		String url="http://study-perf.qa.netease.com/fgadmin/orders/submit";
		JSONObject submit = new JSONObject();
		submit.element("skuIds", "1");
		submit.element("receiverName", "李四");
		submit.element("cellPhone", "20000000004");
		submit.element("addressDetail", "南京大学");		
		submit.element("province", "江苏省");
		submit.element("city", "南京市");
		submit.element("area", "鼓楼区");		
		submit.element("transportFee", 7.0);

		String result=HtmlUnit.doPost(url, submit);
		System.out.println(result);
	}
}

File: QingGuo_request/Request_Demo/Scene5.py
import unittest
from QingGuo.Utils.Sendhttp import SendHttp
from QingGuo.Utils import Common
#作者：刘琪
class Scence5(unittest.TestCase):
    def test_login_success(self):
        self.url = "/common/fgadmin/login"
        user={"phoneArea":"86","phoneNumber":"20000000000","password":"netease123"}
        result=SendHttp().run_http(self.url, "POST", user)
        print ( result )
    def test_address(self):
        self.url = "/fgadmin/address/list"
        user = {"phoneArea": "86", "phoneNumber": "20000000000", "password": "netease123"}
        result = SendHttp().sent_get(self.url,user)
        print(result)
    def test_addressnew_suceess(self):
        self.url = "/fgadmin/address/new"
        addressnew = {"receiverName":"张三",
                      "cellPhone":"12345678901",
                      "addressDetail":"浙江大学",
                      "province":"浙江省",
                      "city":"杭州市",
                      "area":"滨江区"}
        result = SendHttp().send_post_bycookies( self.url,Common.getcookies(addressnew), addressnew )
        print(result)
    def test_getTransportFee_by_address_success(self):
        self.url = "/common/getTransportFee"
        address = {"id":1,"addressDetail":"浙江省_杭州市_滨江区"}
        result = SendHttp().sent_get(self.url, address)
        print(result)
    def test_submit_success(self):
        self.url = "/fgadmin/orders/submit"
        sub={"skuIds":"1",
             "receiverName":"李四",
             "cellPhone":"20000000004",
             "addressDetail":"南京大学",
             "province":"江苏省",
             "city":"南京市",
             "area":"鼓楼区",
             "transportFee":"7.0"}
        result=SendHttp().run_http(self.url,"POST",sub)
        print(result)
if __name__=='__main__':
    unittest.main()